src/Lounge.Web/Controllers/PlayersController.cs

```diff
@@ -250,6 +250,192 @@ public async Task<ActionResult<NameChangeListViewModel>> GetAllPendingNameChange
+                if (currRank != MmrRank)
+                {
+                    if (currRank != null)
+                    {
+                        divisionData.Add(new StatsViewModel.Division
+                        {
+                            Tier = currRank,
+                            Count = tierCount
+                        });
+                    }
+                    currRank = MmrRank;
+                    tierCount = 0;
+                }
+                tierCount++;
+
+                if (player.CountryCode != null)
+                {
+                    if (!countryData.ContainsKey(player.CountryCode))
+                    {
+                        countryData.Add(player.CountryCode, new StatsViewModel.Country
+                        {
+                            PlayerTotal = 0,
+                            TotalAverageMmr = 0,
+                            TopSixMmr = 0,
+                            TopSixPlayers = new List<StatsViewModel.Player>()
+                        });
+                    }
+                    countryData[player.CountryCode].PlayerTotal += 1;
+                    countryData[player.CountryCode].TotalAverageMmr += player.Mmr ?? 0;
+                    if (countryData[player.CountryCode].PlayerTotal <= 6)
+                    {
+                        countryData[player.CountryCode].TopSixMmr += player.Mmr ?? 0;
+                        countryData[player.CountryCode].TopSixPlayers.Add(new StatsViewModel.Player
+                        {
+                            Name = player.Name,
+                            Mmr = player.Mmr ?? 0
+                        });
+                    }
+                }
+            }
+
+            var median = 0;
+
+            if (players.Count > 0)
+            {
+                divisionData.Add(new StatsViewModel.Division
+                {
+                    Tier = currRank,
+                    Count = tierCount
+                });
+
+                foreach (var (key, value) in countryData)
+                {
+                    countryData[key].TotalAverageMmr = value.TotalAverageMmr / value.PlayerTotal;
+                    var topDivisor = value.PlayerTotal < 6 ? value.PlayerTotal : 6;
+                    countryData[key].TopSixMmr = value.TopSixMmr / topDivisor;
+                }
+
+                int mid = players.Count / 2;
+                if (players.Count % 2 != 0)
+                {
+                    if (players[mid].Mmr != null)
+                    {
+                        median = players[mid].Mmr.Value;
+                    }
+                }
+                else
+                {
+                    if (players[mid].Mmr != null)
+                    {
+                        median = (players[mid].Mmr.Value + players[mid - 1].Mmr.Value) / 2;
+                    }
+                }
+
+                AverageMmr = mmrTotal / players.Count;
+            }
+
+            var tables = await _context.Tables
```

Comment:
We should make use of the DbCache for this, we have too many tables currently and this query will take a long time to run.

Reply:
[https://github.com/VikeMK/Lounge-API/pull/39/commits/f6ccd8464f0f81ea498a736a40a32ca1e08b4bc6](url)